src/ert/config/ert_config.py

```diff
@@ -542,17 +544,24 @@ def manifest_to_json(self, iens: int = 0, iter: int = 0) -> Dict[str, Any]:
                     )
                     manifest[name] = file_path
         # Add expected response files to manifest
-        for name, respons_config in self.ensemble_config.response_configs.items():
-            input_file = str(respons_config.input_file)
-            if isinstance(respons_config, SummaryConfig):
-                input_file = input_file.replace("<IENS>", str(iens))
-                manifest[f"{name}_UNSMRY"] = f"{input_file}.UNSMRY"
-                manifest[f"{name}_SMSPEC"] = f"{input_file}.SMSPEC"
-            if isinstance(respons_config, GenDataConfig):
-                if respons_config.report_steps:
-                    for step in respons_config.report_steps:
+        if "summary" in self.ensemble_config.response_configs:
+            smry_config = self.ensemble_config.response_configs["summary"]
+            input_file = smry_config.input_files[0].replace("<IENS>", str(iens))
+            manifest["summary_UNSMRY"] = f"{input_file}.UNSMRY"
```

Comment:
Can be done. I think we should consider, instead of replacing `input_files`, introducing another `@property` where the config tells the outside world what files it is expecting. If we store it "directly", for gen data we'll have to map it back to report steps in read_from_file. If we serialize a list of input_files where the report step is at the end we might get some edge cases where there is a _0 but not from a report step and just a file name, and we can't know how.

Manifest would then look something like this:
```
        for config in self.ensemble_config.response_configs.values():
            for expected_runpath_file in config.expected_input_files:
                manifest[f"{config.response_type}_{expected_runpath_file}"] = (
                    expected_runpath_file
                )
```

Reply:
That looks a lot cleaner and reduces the need of intricate knowledge of each config type